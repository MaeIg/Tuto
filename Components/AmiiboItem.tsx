/* eslint-disable prettier/prettier */
// Components/AmiiboItem

import React from 'react';
import { StyleSheet, View, Text, Image } from 'react-native';

interface AmiiboLayout {
    id: string;
    amiiboSeries: string;
    character: string;
    gameSeries: string;
    name: string;
    image: string;
    release_eu: string;
}

export default function AmiiboItem(props : {amiibo : AmiiboLayout}) {
    const amiibo : AmiiboLayout = props.amiibo;

    return (
        <View style={styles.mainContainer}>
            <Image style={styles.image} source={{uri: amiibo.image}} />
            <View style={styles.contentContainer}>
                <View style={styles.titleContainer}>
                    <Text style={styles.titleText}>{amiibo.name}</Text>
                    <Text style={styles.amiiboSeries}>{amiibo.amiiboSeries}</Text>
                </View>
                <View style={styles.descriptionContainer}>
                    <Text style={styles.descriptionText}>{amiibo.character} from {amiibo.gameSeries} game series !</Text>
                </View>
                <View style={styles.releaseContainer}>
                    <Text style={styles.releaseText}>Release date (eu) : {amiibo.release_eu}</Text>
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    mainContainer: {
        height: 160,
        flex: 1,
        flexDirection: 'row',
        marginLeft: 5,
        marginTop: 5,
    },
    image: {
        height: 150,
        width: 90,
    },
    contentContainer: {
        flex: 1,
        margin: 10,
    },
    titleContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-start',
        flexWrap: 'wrap',
        alignContent: 'flex-start',
    },
    titleText: {
        fontSize: 30,
        fontWeight: 'bold',
        marginRight: 10,
        textAlignVertical: 'bottom',
    },
    amiiboSeries: {
        fontSize: 20,
        fontStyle: 'italic',
        marginBottom: 3,
        textAlignVertical: 'bottom',
    },
    descriptionContainer: {
        flex : 1,
        justifyContent: 'center',
    },
    descriptionText: {
        fontSize: 17,
    },
    releaseContainer: {
        flex: 0.5,
        justifyContent: 'flex-end',
        alignItems: 'flex-end',
    },
    releaseText: {
        fontSize: 16,
    },
});
