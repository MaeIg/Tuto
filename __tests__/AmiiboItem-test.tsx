import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AmiiboItem from '../Components/AmiiboItem';

describe('AmiiboItem', () => {
  const item = {
    head: 'ecdzare',
    tail: 'vsfrdet',
    amiiboSeries: 'amiibo Series',
    character: 'character',
    gameSeries: 'game series',
    name: 'name',
    image:
      'https://raw.githubusercontent.com/N3evin/AmiiboAPI/master/images/icon_00000000-00000002.png',
    release: {
      eu: '2020-03-12',
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<AmiiboItem amiibo={item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
