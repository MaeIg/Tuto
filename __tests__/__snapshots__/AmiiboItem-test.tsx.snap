// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmiiboItem renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "height": 160,
      "marginLeft": 5,
      "marginTop": 5,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://raw.githubusercontent.com/N3evin/AmiiboAPI/master/images/icon_00000000-00000002.png",
      }
    }
    style={
      Object {
        "height": 150,
        "width": 90,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "flex-start",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "marginRight": 10,
            "textAlignVertical": "bottom",
          }
        }
      >
        name
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontStyle": "italic",
            "marginBottom": 3,
            "textAlignVertical": "bottom",
          }
        }
      >
        amiibo Series
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 17,
          }
        }
      >
        character
         from 
        game series
         game series !
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.5,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Release date (eu) : 
        2020-03-12
      </Text>
    </View>
  </View>
</View>
`;
